pre/remap_restart/remap_questions.py

```diff
@@ -13,383 +13,389 @@
 import shutil
 import questionary
 import glob
+from remap_utils import *
 
-def fvcore_name(x):
-  ymdh = x['input:shared:yyyymmddhh']
-  time = ymdh[0:8] + '_'+ymdh[8:10]
-  rst_dir = x.get('input:shared:rst_dir')
-  if not rst_dir : return False
-  files = glob.glob(rst_dir+'/*fvcore_*'+time+'*')
-  if len(files) ==1 :
-    fname = files[0]
-    print('\nFound ' + fname)
-    return fname
+def remove_ogrid_comment(x, opt):
+  ogrid = ''
+  if opt == "IN":
+    ogrid = x.get('input:shared:ogrid')
   else:
-    fname = rst_dir+'/fvcore_internal_rst'
-    if os.path.exists(fname):
-       print('\nFound ' + fname)
-       return fname
-    return False
-
-def tmp_merra2_dir(x):
-   tmp_merra2 = x['output:shared:out_dir']+ '/merra2_tmp_'+x['input:shared:yyyymmddhh']+'/'
-   return tmp_merra2
-
-def data_ocean_default(resolution):
-   default_ = 'CS'
-   if resolution in ['C12','C24', 'C48'] : default_ = '360X180'
-   return default_
-
-def we_default(tag):
-   default_ = '26'
-   if tag in ['INL','GITNL', '525'] : default_ = '13'
-   if tag in ['NL3','NL4', 'NL5', 'v06','v07', 'v08', 'v09'] : default_ = '13'
-   return default_
-
-def zoom_default(x):
-   zoom_ = '8'
-   fvcore = fvcore_name(x)
-   if fvcore :
-      fvrst = os.path.dirname(os.path.realpath(__file__)) + '/fvrst.x -h '
-      cmd = fvrst + fvcore
-      print(cmd +'\n')
-      p = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE)
-      (output, err) = p.communicate()
-      p_status = p.wait()
-      ss = output.decode().split()
-      x['input:shared:agrid'] = "C"+ss[0] # save for air parameter
-      lat = int(ss[0])
-      lon = int(ss[1])
-      if (lon != lat*6) :
-         sys.exit('This is not a cubed-sphere grid fvcore restart. Please contact SI team')
-      ymdh  = x.get('input:shared:yyyymmddhh')
-      ymdh_ = str(ss[3]) + str(ss[4])[0:2]
-      if (ymdh_ != ymdh) :
-         print("Warning: The date in fvcore is different from the date you input\n")
-      zoom = lat /90.0
-      zoom_ = str(int(zoom))
-      if zoom < 1 : zoom_ = '1'
-      if zoom > 8 : zoom_ = '8'
-   if x['input:shared:MERRA-2'] :
-      zoom_ = '2'
-   return zoom_
-
-def get_account():
-   cmd = 'id -gn'
-   p = subprocess.Popen(shlex.split(cmd), stdout=subprocess.PIPE)
-   (accounts, err) = p.communicate()
-   p_status = p.wait()
-   accounts = accounts.decode().split()
-   return accounts[0]
+    ogrid = x.get('output:shared:ogrid')
+  if not ogrid: return False
+
+  ogrid = ogrid.split()[0]
+  if opt == "IN":
+    if ogrid == 'CS':
+       ogrid = x['input:shared:agrid']
+    x['input:shared:ogrid'] = ogrid
+  else:
+    if ogrid == 'CS':
+       ogrid = x['output:shared:agrid']
+    x['output:shared:ogrid'] = ogrid
+  
+  return False
+
+def echo_level(x):
+  if x["output:air:nlevel"] != "72":
```

Comment:
@biljanaorescanin, @weiyuan-jiang: Is this if condition general enough? If I'm reading it correctly, the assumption here is that the number of atm levels in the **input** restarts is always 72.  What if somebody wants to remap from, say, 131 to 181 levels?   Can we just replace the "72" with "input:air:nlevel"??   (Also in line 42.)  Not sure if I understand this correctly.

Reply:
With last commit @weiyuan-jiang fixed this so its more flexible.